File: src/main/org/nlogo/prim/etc/_hidelink.java
package org.nlogo.prim.etc;

import org.nlogo.agent.Link;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.api.Syntax;

public final strictfp class _hidelink
    extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax("---L", true);
  }

  @Override
  public void perform(final Context context) {
    ((Link) context.agent).hidden(true);
    context.ip = next;
  }
}

File: src/main/prim/etc/_hidelink.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.Syntax
import org.nlogo.agent.Link
import org.nlogo.nvm.{ Command, Context }

class _hidelink extends Command {
  override def syntax =
    Syntax.commandSyntax(
      agentClassString = "---L",
      switches = true)
  override def perform(context: Context) {
    context.agent.asInstanceOf[Link].hidden(true)
    context.ip = next
  }
}
